src/lib/mac-address.ts

```diff
@@ -0,0 +1,49 @@
+import * as _ from 'lodash';
+import * as fs from 'mz/fs';
+import * as path from 'path';
+
+import { shouldReportInterface } from '../network';
+import log from './supervisor-console';
+
+export class MacAddressError extends Error {}
+
+export async function getAll(sysClassNet: string): Promise<string | undefined> {
+	try {
+		// read the directories in the sysfs...
+		const interfaces = await fs.readdir(sysClassNet);
+
+		return _(
+			await Promise.all(
+				interfaces.map(async (intf) => {
+					try {
+						const [addressFile, typeFile, masterFile] = [
+							'address',
+							'type',
+							'master',
+						].map((f) => path.join(sysClassNet, intf, f));
+						const intfType = (await fs.readFile(typeFile, 'utf8')).trim();
+						const intfHasMaster = await fs.exists(masterFile);
```

Comment:
Can you either do these fs accesses in parallel or make use of short-circuiting to avoid the second access if the first is invalid